src/vanilla.ts

```diff
@@ -31,6 +31,12 @@ export type StateCreator<T extends State, CustomSetState = SetState<T>> = (
   get: GetState<T>,
   api: StoreApi<T>
 ) => T
+interface AsyncStorage {
+  getItem: (key: string, ...rest: any[]) => Promise<string | null>
+  setItem: (key: string, value: string, ...rest: any[]) => void
+  [x: string]: any
+}
+export type StateStorage = Storage | AsyncStorage
```

Comment:
I don't think we need to define it here, do we?

I prefer more compact type that we really use.
```ts
type StateStorage = {
  getItem: (key: string) => string | null | Promise<string | null>
  setItem: (key: string, value: string) => void | Promise<void>
}
```